src/jaxsim/rbda/contacts/relaxed_rigid.py

```diff
@@ -307,38 +307,42 @@ def compute_contact_forces(
         # collidable points.
         W_H_C = js.contact.transforms(model=model, data=data)
 
-        BW_ν = data.generalized_velocity()
-
-        BW_ν̇_free = jnp.hstack(
-            js.ode.system_acceleration(
-                model=model,
-                data=data,
-                link_forces=references.link_forces(model=model, data=data),
-                joint_torques=references.joint_force_references(model=model),
+        with (
+            data.switch_velocity_representation(VelRepr.Mixed),
+            references.switch_velocity_representation(VelRepr.Mixed),
+        ):
+            BW_ν = data.generalized_velocity()
+
+            BW_ν̇_free = jnp.hstack(
+                js.ode.system_acceleration(
+                    model=model,
+                    data=data,
+                    link_forces=references.link_forces(model=model, data=data),
+                    joint_torques=references.joint_force_references(model=model),
+                )
             )
-        )
 
-        M = js.model.free_floating_mass_matrix(model=model, data=data)
+            M = js.model.free_floating_mass_matrix(model=model, data=data)
 
-        Jl_WC = jnp.vstack(
-            jax.vmap(lambda J, δ: J * (δ > 0))(
-                js.contact.jacobian(model=model, data=data)[:, :3, :], δ
+            Jl_WC = jnp.vstack(
+                jax.vmap(lambda J, δ: J * (δ > 0))(
+                    js.contact.jacobian(model=model, data=data)[:, :3, :], δ
+                )
             )
-        )
 
-        J̇_WC = jnp.vstack(
-            jax.vmap(lambda J̇, δ: J̇ * (δ > 0))(
-                js.contact.jacobian_derivative(model=model, data=data)[:, :3], δ
-            ),
-        )
+            J̇_WC = jnp.vstack(
+                jax.vmap(lambda J̇, δ: J̇ * (δ > 0))(
+                    js.contact.jacobian_derivative(model=model, data=data)[:, :3], δ
+                ),
+            )
 
-        # Compute the regularization terms.
-        a_ref, R, *_ = self._regularizers(
-            model=model,
-            position_constraint=position_constraint,
-            velocity_constraint=velocity,
-            parameters=model.contacts_params,
-        )
+            # Compute the regularization terms.
+            a_ref, R, *_ = self._regularizers(
+                model=model,
+                position_constraint=position_constraint,
+                velocity_constraint=velocity,
+                parameters=model.contacts_params,
+            )
```

Comment:
I don't think the switch has any effects on this
```suggestion
        # Compute the regularization terms.
        a_ref, R, *_ = self._regularizers(
            model=model,
            position_constraint=position_constraint,
            velocity_constraint=velocity,
            parameters=model.contacts_params,
        )
```